hedera-node/src/main/java/com/hedera/services/grpc/marshalling/AliasResolver.java

```diff
@@ -225,24 +223,70 @@ private Result resolveInternalFungible(
             }
             final var resolution = aliasManager.lookupIdBy(alias);
             if (resolution == MISSING_NUM) {
-                result = netOf(isEvmAddress, alias);
+                if (isForToken) {
+                    result = netOf(isEvmAddress, alias, false);
+                } else {
+                    result = netOf(isEvmAddress, alias, true);
+                }
             } else {
                 resolvedAdjust =
                         adjust.toBuilder().setAccountID(resolution.toGrpcAccountId()).build();
             }
             resolutions.put(alias, resolution);
+            tokenTransferResolutions.put(alias, resolution);
         }
         resolvingAction.accept(resolvedAdjust);
         return result;
     }
 
-    private Result netOf(final boolean isEvmAddress, final ByteString alias) {
+    private void perceiveNftReceiverResult(final Result result, final NftTransfer change) {
+        perceiveReceiverResult(
+                result, change.getSerialNumber(), change.getReceiverAccountID().getAlias(), false);
+    }
+
+    private void perceiveNonNftReceiverResult(final Result result, final AccountAmount adjust) {
+        perceiveReceiverResult(result, adjust.getAmount(), adjust.getAccountID().getAlias(), true);
+    }
+
+    private void perceiveReceiverResult(
```

Comment:
Oops I shouldn't have suggested `perceveReceiverResult`, since this is also called for non-NFT "sender"s. 😝 

Maybe just `perceiveResult()`.

Reply:
Agreed. Fixed name to `perceiveResult()` and `perceiveNonNftResult`